src/test/denialofservice_tests.cpp

```diff
@@ -34,6 +36,56 @@ static CService ip(uint32_t i)
 
 BOOST_FIXTURE_TEST_SUITE(denialofservice_tests, TestingSetup)
 
+void ConnectNode(ConnmanTestMsg& connman, PeerManager& peerman, CNode& node, ServiceFlags their_services) noexcept
+{
+    // Send version message
+    const CNetMsgMaker mm{0};
+    CSerializedNetMsg msg_version{
+        mm.Make(NetMsgType::VERSION,
+                PROTOCOL_VERSION,                               // current p2p version
+                Using<CustomUintFormatter<8>>(their_services),  // their service flags
+                int64_t{},                                      // dummy time
+                int64_t{},                                      // ignored service bits
+                CService{},                                     // dummy
+                int64_t{},                                      // ignored service bits
+                CService{},                                     // ignored
+                uint64_t{1},                                    // dummy nonce
+                std::string{},                                  // dummy subver
+                int32_t{},                                      // dummy starting_height
+                true),                                          // fRelay
+    };
+
+    (void)connman.ReceiveMsgFrom(node, msg_version);
+    node.fPauseSend = false;
+    connman.ProcessMessagesOnce(node);
+
+    // Receive messages
+    {
+        LOCK(node.cs_sendProcessing);
+        peerman.SendMessages(&node);
+    }
+    if (node.fDisconnect) return;
+
+    BOOST_CHECK_EQUAL(node.nVersion, PROTOCOL_VERSION);
+    BOOST_CHECK_EQUAL(node.GetCommonVersion(), PROTOCOL_VERSION);
+    BOOST_CHECK_EQUAL(node.nServices, their_services);
+
+    // Send verack message
+    CSerializedNetMsg msg_verack{mm.Make(NetMsgType::VERACK)};
+    (void)connman.ReceiveMsgFrom(node, msg_verack);
+    node.fPauseSend = false;
+    connman.ProcessMessagesOnce(node);
+
+    // Receive messages
+    {
+        LOCK(node.cs_sendProcessing);
+        peerman.SendMessages(&node);
+    }
+    BOOST_CHECK_EQUAL(node.fSuccessfullyConnected, true);
+
+    TestOnlyResetTimeData();
```

Comment:
The Ci was failing because the peer manager calls `AddTimeData`  when receiving a version message and since the time offset data for adjusted time is a global this can mess with other tests. Calling `TestOnlyResetTimeData()` here should fix it.